### PR TITLE
[scheduler] Add postsubmit subscription

### PR DESCRIPTION
**Pull request by CaseyHillers**
https://github.com/flutter/flutter/issues/81795

### CODE REVIEW

**Comment by CaseyHillers** [comment]
Friendly ping now that the code freeze is over

Friendly ping

File: app_dart/lib/src/request_handlers/postsubmit_luci_subscription.dart [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  @override
+  Future<Body> post() async {
+    final ClientContext clientContext = authContext!.clientContext;
+    final DatastoreService datastore = datastoreProvider(config.db);
+    final KeyHelper keyHelper = KeyHelper(applicationContext: clientContext.applicationContext);
+
+    final String data = (await message)!.data!;
+    final BuildPushMessage buildPushMessage =
+        BuildPushMessage.fromJson(json.decode(String.fromCharCodes(base64.decode(data))) as Map<String, dynamic>);
+    log.fine(buildPushMessage.userData);
```
Comment (keyonghan) [inline app_dart/lib/src/request_handlers/postsubmit_luci_subscription.dart:48]: Nit: add an example of `userData`.

File: app_dart/test/src/request_handling/fake_authentication.dart [hunk 2]
```diff
@@ -94,7 +93,7 @@ class FakeClientContext implements ClientContext {
 
 class FakeAppEngineContext implements AppEngineContext {
   @override
-  late String applicationID;
+  late String applicationID = 'flutter-dashboard';
```
Comment (keyonghan) [inline app_dart/test/src/request_handling/fake_authentication.dart:96]: Do we still need `late`?

File: app_dart/test/src/request_handling/fake_authentication.dart [hunk 3]
```diff
@@ -112,7 +111,7 @@ class FakeAppEngineContext implements AppEngineContext {
   late String module;
 
   @override
-  late String partition;
+  late String partition = '[default]';
```
Comment (keyonghan) [inline app_dart/test/src/request_handling/fake_authentication.dart:114]: Ditto.

File: app_dart/lib/src/request_handlers/postsubmit_luci_subscription.dart [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import '../request_handling/exceptions.dart';
+import '../request_handling/subscription_handler.dart';
+import '../service/config.dart';
+import '../service/datastore.dart';
+import '../service/logging.dart';
+
+/// An endpoint for listening to build updates for postsubmit builds.
+///
+/// The PubSub subscription is set up here:
+/// https://cloud.google.com/cloudpubsub/subscription/detail/luci-postsubmit?project=flutter-dashboard&tab=overview
```
Comment (keyonghan) [inline app_dart/lib/src/request_handlers/postsubmit_luci_subscription.dart:25]: Where can I take a look at the messages? Used the url with pantheon, but no messages are available?

Comment (CaseyHillers) [reply app_dart/lib/src/request_handlers/postsubmit_luci_subscription.dart:25]: PubSub messages aren't available in GCP. You can view them in the logs by searching for the handlers: `/api/presubmit-luci-subscription`

File: app_dart/lib/src/request_handlers/postsubmit_luci_subscription.dart (hunk 1 above)
Comment (CaseyHillers) [reply app_dart/lib/src/request_handlers/postsubmit_luci_subscription.dart:48]: Added

File: app_dart/test/src/request_handling/fake_authentication.dart (hunk 2 above)
Comment (CaseyHillers) [reply app_dart/test/src/request_handling/fake_authentication.dart]: Nope, thanks for catching this!

**Review by keyonghan — APPROVED** [review]
As the task status update is still on from framework test runner, I believe this PR is safe to land.

LGTM.

**Comment by CaseyHillers** [comment]
> As the task status update is still on from framework test runner, I believe this PR is safe to land.

Yep! This is a no-op. I have 2 PRs after this to schedule cocoon based targets in the scheduler class. Then we can start migrating the targets in the ci.yaml.

**Pull request merged**